cdap-api-common/src/main/java/co/cask/cdap/api/data/schema/Schema.java

```diff
@@ -204,7 +222,38 @@ public static Schema of(Type type) {
     if (!type.isSimpleType()) {
       throw new IllegalArgumentException("Type " + type + " is not a simple type.");
     }
-    return new Schema(type, null, null, null, null, null, null, null);
+    return new Schema(type, null, null, null, null, null, null, null, null);
+  }
+
+  /**
+   * Creates a {@link Schema} for the given logical type. The logical type given must be a
+   * {@link Schema.LogicalType}.
+   * @param logicalType LogicalType of the schema to create.
+   * @return A {@link Schema} with the given logical type.
+   */
+  public static Schema of(LogicalType logicalType) {
+    Type type;
+    switch (logicalType) {
+      case DATE:
+        type = Type.INT;
+        break;
+      case TIMESTAMP_MILLIS:
+        type = Type.LONG;
+        break;
+      case TIMESTAMP_MICROS:
+        type = Type.LONG;
+        break;
+      case TIME_MILLIS:
+        type = Type.INT;
+        break;
+      case TIME_MICROS:
+        type = Type.LONG;
+        break;
+      default:
+        throw new IllegalStateException("LogicalType " + logicalType + " is not a supported logical type.");
```

Comment:
Should be an `IllegalArgumentException`

Reply:
This was modified from `IllegalArgumentException` to `IllegalStateException `  as per the comment from @albertshau https://github.com/caskdata/cdap/pull/10549#discussion_r211032536

Reply:
the reasoning is because the input LogicalType should always be valid. If we get here, it's not the fault of the caller, but the fault of this class for not handling all enum values.

Reply:
Actually in this case, it is better to move this logic inside the enum so that there won’t be any missing case even when new enum is added

Reply:
By doing so, no exception would ever be raised

Reply:
It is mentioned in effective java

Reply:
Makes sense, thanks for the suggestion!